### PR TITLE
Add instructions for feeding/migrating to production databases and backup notes

### PR DESCRIPTION
**Pull request by aramalipoor**
I needed such a section in database docs when I used AbarCloud in my early days.

It might not be a good thing to share such snippets, so WDYT? Should we keep this?

### CODE REVIEW

**Comment by alikhajeh1** [comment]
@aramalipoor Good idea! this made me re-think our MySQL/PG docs so the quickstart has all the needed docs for non-production (as that's what users start with), and the additional services pages have production docs: https://github.com/abarcloud/docs/pull/43

Can you plz review/merge that PR then add a section in those pages for users to upload their dumps? I wonder if we should provide generic instructions (not with docker) as most users won't have docker on their existing servers.

For PG, since they can't connect to it directly, they'll probably need to create a non-production pod first, add a storage volume, rsync their file to that, then from that pod, import to their production PG.

File: quickstart/database.md [hunk 1]
```diff
@@ -61,3 +61,46 @@ The AbarCloud CLI needs to be used to upload files to pods.
...
   psql MY_DATABASE_NAME < /tmp/db_dump/MY_FILE.SQL
   ```
+
+## Feed/Migrate to Production Databases
+To feed your initial production database or migrate from a previous provider you can connect to your instance using the external hostname given to you by AbarCloud.
+
+Here is a snippet which uses a one-off Docker container to upload a `.sql` dump unto your production database.
+```sh
+# Information below are provided when you've ordered a MySql Production instance.
+DATABASE_EXTERNAL_HOST=mysql-xxxxxx.ext.eu-central-1a.abarcloud.com # Provided by AbarCloud
```
Comment (alikhajeh1) [inline quickstart/database.md]: we try to avoid making endpoints public, users are given different endpoints...

**Comment by aramalipoor** [comment]
I've made the changes so now users know:

* How to **initialize** the database from a dump on their local machine.
* How to **backup** the database to their local machine.

@alikhajeh1 if looks good to you we can merge :+1:

File: additional-services/mysql.md [hunk 2]
```diff
@@ -35,6 +35,29 @@ We do not recommend the use of the root database user in your applications. Inst
   GRANT ALL PRIVILEGES ON my_new_database.* TO 'my-new-user'@'%';
   ```
 
+## Initialize Database
+To feed your **initial production** database or **migrate from a previous provider** you can connect to your instance using the external hostname given to you by AbarCloud.
+
+Here is a snippet to upload a `.sql` dump unto your production database.
```
Comment (alikhajeh1) [inline additional-services/mysql.md]: unto vs into? maybe into?

File: additional-services/mysql.md [hunk 3]
```diff
@@ -35,6 +35,29 @@ We do not recommend the use of the root database user in your applications. Inst
...
+```sh
+BACKUP_FILE=/home/USER/PROJECT/backups/xxxxxxxx.sql
```
Comment (alikhajeh1) [inline additional-services/mysql.md]: The notation `/FULL/PATH/TO/FOLDER` elsewhere

File: additional-services/mysql.md [hunk 4]
```diff
@@ -35,6 +35,29 @@ We do not recommend the use of the root database user in your applications. Inst
...
+    -D MY_DATABASE_NAME \
+    -e "source ${BACKUP_FILE}"
+```
+
 ## Slow queries
 
 Run `SELECT * FROM mysql.slow_log_view` to check for queries that took longer than 1s to return.
+
+## Backup on your Local Machine
+In early stages of your application, when production database is just a few hundred megabytes, you might want to have production data on your development machine.
```
Comment (alikhajeh1) [inline additional-services/mysql.md]: If your production database is small, you can copy the database to your local machine using the following command:

File: additional-services/mysql.md [hunk 5]
```diff
@@ -35,6 +35,29 @@ We do not recommend the use of the root database user in your applications. Inst
...
+```
+
 ## Slow queries
 
 Run `SELECT * FROM mysql.slow_log_view` to check for queries that took longer than 1s to return.
+
+## Backup on your Local Machine
+In early stages of your application, when production database is just a few hundred megabytes, you might want to have production data on your development machine. 
+
+Here is a snippet which runs a `mysqldump` on your production database.
```
Comment (alikhajeh1) [inline additional-services/mysql.md]: If you do use the above, this line can go

File: additional-services/mysql.md [hunk 6]
```diff
@@ -35,6 +35,29 @@ We do not recommend the use of the root database user in your applications. Inst
...
+    --user=DB_USER --password=DB_PASSWORD \
+    -D MY_DATABASE_NAME \
+    -e "source ${BACKUP_FILE}"
+```
+
 ## Slow queries
 
 Run `SELECT * FROM mysql.slow_log_view` to check for queries that took longer than 1s to return.
+
+## Backup on your Local Machine
```
Comment (alikhajeh1) [inline additional-services/mysql.md]: on second thoughts, it's not great security practice to store production DBs on local machines, should we just take this out and users who really want it, can figure it out themselves?

File: additional-services/postgres.md [hunk 7]
```diff
@@ -44,11 +44,31 @@ We do not recommend the use of the root database user in your applications. Inst
   GRANT ALL PRIVILEGES ON DATABASE my_new_database TO my_new_user;
   ```
 
+## Initialize Database
+To feed your **initial production** database or **migrate from a previous provider** you can connect to your instance using the external hostname given to you by AbarCloud.
```
Comment (alikhajeh1) [inline additional-services/postgres.md]: PG doesn't have external hostname.

File: additional-services/postgres.md [hunk 8]
```diff
@@ -44,11 +44,31 @@ We do not recommend the use of the root database user in your applications. Inst
...
+
+1. If you have a **large database** (more than a few hundred megabytes) create a [volume storage](../management/volume-storage.md).
```
Comment (alikhajeh1) [inline additional-services/postgres.md]: storage volume (both for text and link)

File: additional-services/postgres.md [hunk 9]
```diff
@@ -44,11 +44,31 @@ We do not recommend the use of the root database user in your applications. Inst
...
+2. Then create a temporary non-production Postgres pod, as per **[instructions here](postgres.md#connection)**.
```
Comment (alikhajeh1) [inline additional-services/postgres.md]: Create a temporary [non-production PostgreSQL](link) deployment.

One of my uni lecturers had issues with people not using hyperlinks as they were meant to be used (i.e. inline for text, vs "click here") so he drilled that into my head lol

File: additional-services/postgres.md [hunk 10]
```diff
@@ -44,11 +44,31 @@ We do not recommend the use of the root database user in your applications. Inst
...
+    * If you have created a volume storage bind it to your pod's `/data`
```
Comment (alikhajeh1) [inline additional-services/postgres.md]: If you have created a storage volume, attach it to PostgreSQL's `/data` directory.

File: additional-services/postgres.md [hunk 11]
```diff
@@ -44,11 +44,31 @@ We do not recommend the use of the root database user in your applications. Inst
...
+3. Get your temp Postgres pod name using `oc get pods` and run following command:
```
Comment (alikhajeh1) [inline additional-services/postgres.md]: temp - temporary
Postgres -> PostgreSQL

File: quickstart/database.md [hunk 12]
```diff
@@ -11,7 +11,7 @@ You can go to Applications > Deployments > your application and click on the Env
 2. Monitored automatically: you'll get notified if the CPU, memory, disk, IOPS, or connection thresholds are reached.
 3. Backed-up automatically: every night, kept for 7 days.
 
-## Upload existing DB dumps
+## Upload existing database dumps (test & staging)
```
Comment (alikhajeh1) [inline quickstart/database.md]: I'd take out `(test & staging)` as the rest of the page clearly mentions this page is about non-production

File: additional-services/postgres.md [hunk 13]
```diff
@@ -65,3 +85,24 @@ PostgreSQL has the [pg_stat_statements extension](https://www.postgresql.org/doc
   # Remove the extension when you're done as it has a performance penalty since it tracks all queries
   DROP EXTENSION pg_stat_statements;
   ```
+
+## Backup on your Local Machine
```
Comment (alikhajeh1) [inline additional-services/postgres.md]: Same comments as MySQL.

File: additional-services/postgres.md (hunk 9 above)
Comment (aramalipoor) [reply additional-services/postgres.md]: He's right! I have to drill it into my head as well! Any kind of feedback is more valuable than gold for me Ali, thanks for making me rich :1st_place_medal:

File: additional-services/mysql.md (hunk 6 above)
Comment (aramalipoor) [reply additional-services/mysql.md]: That's what I had in mind as well. You're right, let's remove this for both MySql and PostgreSQL :+1:

**Pull request merged**